compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3565,28 +3565,34 @@ class Typer extends Namer
   /** Types the body Scala 2 macro declaration `def f = macro <body>` */
   private def typedScala2MacroBody(call: untpd.Tree)(using Context): Tree =
     // TODO check that call is to a method with valid signature
-    def typedPrefix(tree: untpd.RefTree): Tree = {
+    def typedPrefix(tree: untpd.RefTree)(splice: Context ?=> Tree => Tree)(using Context): Tree = {
       tryAlternatively {
-        typedExpr(tree, defn.AnyType)
+        splice(typedExpr(tree, defn.AnyType))
       } {
         // Try to type as a macro bundle
         val ref = tree match
           case Ident(name) => untpd.Ident(name.toTypeName).withSpan(tree.span)
           case Select(qual, name) => untpd.Select(qual, name.toTypeName).withSpan(tree.span)
         val bundle = untpd.Apply(untpd.Select(untpd.New(ref), nme.CONSTRUCTOR), untpd.Literal(Constant(null))).withSpan(call.span)
```

Comment:
This .withSpan(call.span) should probably be around the tpd.Block

Reply:
perhaps I overdid it?,

Reply:
I removed some of them. Could you check it works with your example?

Reply:
I'm not sure whats happening because the CI is passing but `testCompilation tests/pos/i8945.scala` fails for me

Reply:
wait CI is still running

Reply:
`.withSpan` still needs to be added to `bundleVal`

Reply:
@nicolasstucki the commit I just added also works with my scala 2 testing